File: ud05/e5/src/main/java/com/example/AlumnoManager.java
package com.example;

import java.util.ArrayList;
import java.util.List;

public class AlumnoManager {
    private List<Alumno> alumnos = new ArrayList<>();

    public void agregarAlumno(Alumno a) {
        if (a != null) {
            alumnos.add(a);
        }
    }

    public Alumno buscarAlumnoPorNombre(String nombre) {
        for (Alumno a : alumnos) {
            if (a.getNombre().equals(nombre)) {
                return a;
            }
        }
        return null;
    }

    public double calcularMedia() {
        double suma = 0;
        for (Alumno a : alumnos) {
            suma = suma + a.getNotaMedia();
        }
        if (alumnos.size() == 0) {
            return 0;
        }
        return suma / alumnos.size();
    }

    public String clasificarAlumno(Alumno a) {
        if (a == null) {
            return "Alumno nulo";
        }

        if (a.getNotaMedia() >= 9 && a.getEdad() >= 18) {
            return "Sobresaliente adulto";
        } else if (a.getNotaMedia() >= 9 && a.getEdad() < 18) {
            return "Sobresaliente menor";
        } else if (a.getNotaMedia() >= 7 && a.getEdad() >= 18) {
            return "Notable adulto";
        } else if (a.getNotaMedia() >= 7 && a.getEdad() < 18) {
            return "Notable menor";
        } else if (a.getNotaMedia() >= 5 && a.getEdad() >= 18) {
            return "Aprobado adulto";
        } else if (a.getNotaMedia() >= 5 && a.getEdad() < 18) {
            return "Aprobado menor";
        } else if (a.getEdad() >= 18) {
            return "Suspenso adulto";
        } else {
            return "Suspenso menor";
        }
    }

    public void mostrarAlumnosAprobados() {
        for (Alumno a : alumnos) {
            if (a.getNotaMedia() >= 5) {
                System.out.println("Alumno aprobado: " + a.getNombre());
            }
        }
    }

    public void mostrarAlumnosSuspensos() {
        for (Alumno a : alumnos) {
            if (a.getNotaMedia() < 5) {
                System.out.println("Alumno suspenso: " + a.getNombre());
            }
        }
    }

    public List<Alumno> getAlumnos() {
        return alumnos;
    }

    public void setAlumnos(List<Alumno> alumnos) {
        this.alumnos = alumnos;
    }
}

File: ud05/e6/src/main/kotlin/com/example/AlumnoManager.kt
package com.example

class AlumnoManager {

    private var alumnos: MutableList<Alumno> = mutableListOf()

    fun agregarAlumno(a: Alumno?) {
        if (a != null) {
            alumnos.add(a)
        }
    }

    fun buscarAlumnoPorNombre(nombre: String): Alumno? {
        for (a in alumnos) {
            if (a.nombre == nombre) {
                return a
            }
        }
        return null
    }

    fun calcularMedia(): Double {
        var suma = 0.0

        for (a in alumnos) {
            suma += a.notaMedia
        }

        if (alumnos.isEmpty()) {
            return 0.0
        }

        return suma / alumnos.size
    }

    fun clasificarAlumno(a: Alumno?): String {
        if (a == null) {
            return "Alumno nulo"
        }

        return if (a.notaMedia >= 9 && a.edad >= 18) {
            "Sobresaliente adulto"
        } else if (a.notaMedia >= 9 && a.edad < 18) {
            "Sobresaliente menor"
        } else if (a.notaMedia >= 7 && a.edad >= 18) {
            "Notable adulto"
        } else if (a.notaMedia >= 7 && a.edad < 18) {
            "Notable menor"
        } else if (a.notaMedia >= 5 && a.edad >= 18) {
            "Aprobado adulto"
        } else if (a.notaMedia >= 5 && a.edad < 18) {
            "Aprobado menor"
        } else if (a.edad >= 18) {
            "Suspenso adulto"
        } else {
            "Suspenso menor"
        }
    }

    fun mostrarAlumnosAprobados() {
        for (a in alumnos) {
            if (a.notaMedia >= 5) {
                println("Alumno aprobado: ${a.nombre}")
            }
        }
    }

    fun mostrarAlumnosSuspensos() {
        for (a in alumnos) {
            if (a.notaMedia < 5) {
                println("Alumno suspenso: ${a.nombre}")
            }
        }
    }

    fun getAlumnos(): MutableList<Alumno> {
        return alumnos
    }

    fun setAlumnos(alumnos: MutableList<Alumno>) {
        this.alumnos = alumnos
    }
}